usl_pipeline/cloud_functions/main.py

```diff
@@ -675,24 +675,136 @@ def _compute_custom_wps_variables(dataset: xarray.Dataset) -> xarray.Dataset:
     Returns:
       A new xarray dataset with newly derived variables.
     """
-    # Derive wind components: WSPD10 (wind speed), WDIR10 (wind direction)
-    if all(var in dataset.keys() for var in ["UU", "VV"]):
+    # Derive wind components from UU & VV + Solar time components from XLAT_M & XLONG_M
+   
+    # WSPD (wind speed) at FNL level 0 (~10m)
+    # WDIR_SIN (wind direction_sine component) at FNL level 0 (~10m)
+    # WDIR_COS (wind direction_cosine component) at FNL level 0 (~10m)
+    if all(var in dataset.keys() for var in ["UU", "VV", "XLAT_M", "XLONG_M"]):
```

Comment:
I would recommend splitting out the solar time computation, separated from the wind components computation since the solar time will not depend on UU or VV - so like:

```python
# Derive wind components from UU and VV: WSPD, WDIR
if all(var in dataset.keys() for var in ["UU", "VV"]):
    <computation for wind components only>

# Derive solar time from XLAT_M and XLONG_M
if all(var in dataset.keys() for var in ["XLAT_M", "XLONG_M"]):
    <computation for solar time only>
```

Reply:
Up to you but another thing you can do in addition to this to make this code slightly more readable is to break out the wind and solat time computations into helper methods that have descriptive names (to remove need for comments like `#### Compute WIND ####`

example refractor you can do:
```python
def _compute_custom_wps_variables(dataset: xarray.Dataset) -> xarray.Dataset:
     WSPD, WSPD_SIN, WSPD_COS = _compute_wind_components(dataset)
     dataset = dataset.assign(WSPD=(new_dims, wind_speed))
     dataset = dataset.assign(WDIR_SIN=(new_dims, wind_direction_sin))
     dataset = dataset.assign(WDIR_COS=(new_dims, wind_direction_cos))

      SOLAR_TIME_SIN, SOLAR_TIME_COS = _compute_solar_time_components(dataset)
     dataset = dataset.assign(SOLAR_TIME_SIN=(new_dims, sin_solar_time))
     dataset = dataset.assign(SOLAR_TIME_COS=(new_dims, cos_solar_time))

def _compute_wind_components:
    <all the computations for wind>

def _compute_solar_time_components:
    <all the computations for solar time>
```

Reply:
Maybe for the future. Thanks!!!! :)